src/Sarif/Visitors/InsertOptionalDataVisitor.cs

```diff
@@ -143,5 +145,36 @@ internal FileData VisitDictionaryValueNullChecked(string key, FileData node)
 
             return base.VisitFileData(node);
         }
+
+        public override Result VisitResult(Result node)
+        {
+            _ruleId = node.RuleId;
+            node = base.VisitResult(node);
+            _ruleId = null;
+
+            return node;
+        }
+
+        public override Message VisitMessage(Message node)
+        {
+            if ((node.Text == null || _dataToInsert.Includes(OptionallyEmittedData.OverwriteExistingData)) &&
+                _dataToInsert.Includes(OptionallyEmittedData.FlattenedMessages))
+            {
+                Rule rule = null;
+                //string formatString = null;
+                if ((bool)_run.Resources?.Rules.TryGetValue(_ruleId, out rule))
```

Comment:
>(bool) [](start = 20, length = 6)

This fails at runtime if the `Nullable<bool>` that results from this expression actually is `null`.

This alternative always works:
```
if (_run.Resources?.Rules.TryGetValue(_ruleId, out rule) == true)
```